### DISCUSSION TITLE
Decrementing Item quantity

### DISCUSSION THREAD

**Discussion by v4nt0lin** [Shopify/storefront-api-feedback#222] [Help] [answered]
I'm currently building a NextJS site using the Storefront API and have managed to use 'cartLinesAdd' to increment my product quantity in my cart by one on each click of a plus icon (adding 1 to quantity). At the moment i'm failing to have my minus icon do the opposite (using -1 on quantity) and wondering if there is a specific approach I should be using?

              onClick={async () => {
                await updateCart(cartData.cart.id, [
                  {
                    merchandiseId: merchandise.id,
                    quantity: 1,
                  },
                ])
                await refreshCartData()
              }}
              
To add context my onClick will work when adding 1, but fails when using -1         

Many thanks!

**Comment by johncraigcole** [comment, marked as answer]
Hey @v4nt0lin think of `cartLinesAdd` as being something to be used on a PDP page for that initial add to the cart. For modifications of lines already in the cart,  [cartLinesUpdate](https://shopify.dev/docs/api/storefront/2023-07/mutations/cartlinesupdate) is the mutation to use. In your example, if I click an increment or decrement button when calling the `cartLinesUpdate` mutation you're specifying the new quantity not a delta.
